core/src/main/java/com/transferwise/common/gracefulshutdown/strategies/BaseReactiveResourceShutdownStrategy.java

```diff
@@ -78,40 +120,52 @@ protected int getResourceFullShutdownTimeoutMs() {
   }
 
   /**
-   * {@link Duration} allowed for resource to shut down gracefully. If not shut down within this time -
-   * {@link BaseReactiveResourceShutdownStrategy#shutdownResourceForced} will be called.
-   * @return {@link Duration}
+   * Split ResourceFullShutdownTimeout between graceful and forced shutdown. Value between 0 and 1 is expected where 1 means 100%.
+   * <ul>
+   *   <li>{@link #getResourceGracefulShutdownTimeout} =
+   *   ({@link #getResourceFullShutdownTimeoutMs} -
+   *   {@link #getStrategyShutdownDelay}) *
+   *   {@link #getResourceGracefulShutdownTimeShare}
+   *   </li>
+   *   <li>{@link #getResourceForcedShutdownTimeout} =
+   *   {@link #getResourceFullShutdownTimeoutMs} -
+   *   {@link #getResourceGracefulShutdownTimeout}
+   *   </li>
+   * </ul>
+   *
+   * @return time share of ResourceFullShutdownTimeout dedicated to graceful shutdown
    */
-  protected Duration getResourceGracefulShutdownTimeout() {
-    // use 9/10 of time full shut down time to graceful shutdown
-    return Duration.ofMillis(getResourceFullShutdownTimeoutMs() / 10 * 9);
+  protected double getResourceGracefulShutdownTimeShare() {
+    return 0.9d;
   }
 
   /**
-   * {@link Duration} allowed for resource to force shut down. If not shut down within this time - error will be logged.
+   * {@link Duration} allowed for resource to shut down gracefully. If not shut down within this time -
+   * {@link #shutdownResourceForced} will be called.
+   *
    * @return {@link Duration}
    */
-  protected Duration getResourceForcedShutdownTimeout() {
-    // use 1/10 of time full shut down time to force shutdown
-    return Duration.ofMillis(getResourceFullShutdownTimeoutMs() / 10);
-  }
-
-
-  @Getter(AccessLevel.PROTECTED)
-  private final Scheduler shutdownScheduler = Schedulers.newBoundedElastic(
-      10,
-      100_000,
-      "ShutdownWorker_" + this.getClass().getSimpleName()
-  );
+  private Duration getResourceGracefulShutdownTimeout() {
+    // This checks incorrect configuration. Timeout error will be logged.
+    if (getStrategyShutdownDelay().toMillis() > getResourceFullShutdownTimeoutMs()) {
+      return Duration.ZERO;
+    }
 
-  private final List<T> addedResources = new ArrayList<>();
+    // getStrategyShutdownDelay time of getResourceFullShutdownTimeoutMs was already used.
+    // use ResourceGracefulShutdownTimeShare of remaining time for graceful shutdown
+    double gracefulShutdownShare = getResourceGracefulShutdownTimeShare();
+    if (gracefulShutdownShare > 1d) {
```

Comment:
Probably `Math.min()` looks better in this case.

Reply:
Probably. I personally like both versions: with and without using `Math.min()`. Changed